compiler/rustc_middle/src/mir/pretty.rs

```diff
@@ -1125,6 +1126,9 @@ fn post_fmt_projection(projection: &[PlaceElem<'_>], fmt: &mut Formatter<'_>) ->
             ProjectionElem::OpaqueCast(ty) => {
                 write!(fmt, " as {ty})")?;
             }
+            ProjectionElem::Subtype(ty) => {
+                write!(fmt, "as {ty})")?;
```

Comment:
```suggestion
                write!(fmt, " as subtype {ty})")?;
```
I think it's harder to debug issues if different projections have the dsame formatting.

though I guess, cc @oli-obk whether this affects the `mir!` macro or sth

also, missing space